File: Python/Arreglos/Arreglos-5.py
def main():
    # Calcular el promedo ponderado de X notas
    cantNotasString = input("Ingrese la cantidad de evaluaciones: ")
    cantNotas = int(cantNotasString)
    notas = []
    ponderaciones = []
    suma = 0
    sumaPonderaciones = 0
    for i in range(0, cantNotas):
        notas.append(float(input("Ingrese la nota " + str(i + 1) + ": ")))
        ponderaciones.append(float(input("Ingrese la ponderacion: ")))
        suma = suma + (notas[i] * ponderaciones[i])
        sumaPonderaciones = sumaPonderaciones + ponderaciones[i]

    print("La nota final es: ")
    print(suma / sumaPonderaciones)

main()

File: CPP/Arreglos/Arreglos-5.cpp
#include <iostream>

int main(){
    int canNumeros;
    std::cout<<"Ingrese la cantidad de notas: "<<std::endl;
    std::cin>>canNumeros;

    float notas[canNumeros];
    float ponderaciones[canNumeros];
    float sumaProducto = 0.0, sumaPonderaciones = 0.0;

    float suma = 0.0;

    std::cout<<"Ingrese sus notas y sus ponderaciones"<<std::endl;

    for (int i = 0; i < canNumeros; i++){
        std::cout<<"Ingrese su nota y ponderacion: "<<std::endl;
        std::cin>>notas[i];
        std::cin>>ponderaciones[i];
        sumaProducto = sumaProducto + (notas[i]*ponderaciones[i]);
        sumaPonderaciones = sumaPonderaciones + ponderaciones[i];

    }
    std::cout<<"El promedio es: "<< sumaProducto/ sumaPonderaciones << std::endl;
}
